### ISSUE TITLE
Hostile entities not dying without player being present

### ISSUE THREAD

**Issue by phileaphi** [CompactMods/CompactMachines#337]
**Compact Machines version**:1.12.2-3.0.12-b215
**Forge version**:14.23.4.2705
**I am not using Optifine**: no optifine running
**Screenshot (if possible)**: [here](https://cdn.discordapp.com/attachments/357597633566605315/455493287374225438/unknown.png)

**Description of the problem including expected versus actual behavior:**
Hostile entities killed in a compacting machine without the player present do not die. Obviously one expects a mobfarm to work in a compacting machine.

**Further detail:**
*Is it the storage?:* **No**
While I am present in the compacting machine the mobs get killed and the loot manages to go into my storage drawer system. While I m outside, nothing comes into my storage drawer system but changes from me to the drawer system with AE2 are registered and handled correctly.
*Are mobs spawning while outside?*:  **Yes**
I connected a redstone tunnel to disable/enable the spawner(IF Mob Duplicator) from the outside. After enabling for ~10 seconds and disabling there were mobs inside the compacting machine.
*Is it the type of mob killer(DE or IF)?*: **No**
Both don't kill the mobs so it's highly unlikely to be the other mod's fault.
*Further Observations*:
Using DE's stabilized mobspawners which have no spawnlimit made the server severely lag and the IF's mob crushers had gone up to using 4000ms calculation probably caused by the amount of mobs it tries to kill.

**Steps to reproduce:**
Create a mobfarm inside a compacting machine using a grinder to kill the mobs and wait outside for the mobs to die and wait for the loot...(forever)

**Comment by davenonymous** [comment]
Sounds like a chunk loading issue. Does a chunkloader inside the machine fix your issue?
Is the compact machines dimension being unloaded when you leave it (should be in your logs)?

Need more information to reproduce this - automated mob farms obviously worked when I last tested them.

**Comment by phileaphi** [comment]
I'm playing the FTB Revelations 2.0 pack. Chunk Loading with FTB Utils didn't change it and the dimension doesn't get unloaded otherwise my ore processing would stop too. The chunk is definitely loaded as mobs can spawn in there without a player present.

Edit: You can add me in discord ************

**Comment by davenonymous** [comment]
I'll have to reproduce this; might take some time for me to get around to it, please be patient.

The issue sounds really weird though and should not be happening as you describe.
It's chunk-loaded, but the machines only work with a player present - sounds to me like the machines doing the killing are having issues, not the Compact Machine itself.

I'm not using Discord.

**Comment by phileaphi** [comment]
Would it help if I upload the world this is occuring in?

Edit: Even Diamond Spikes won't work

**Comment by Lodran** [comment]
While playing on the ToastyNetworks servers (which have custom chunkloaders), I noticed that there appear to be two stages for chunkloading - one in which the entities are poorly processed, and another where they get enough processor time to run properly.

When my spawner had no chunkloader, entities inside would move jerkily (once every second or so), as long as I was more than 2 chunks away (The chunk was loaded, and new entities were being produced, but they were starved for time).  If I moved closer in, they would move smoothly.

By adding a (server specific) chunkloader, things started moving smoothly all the time.

Perhaps your chunkloading code isn't upgrading entity processing?
